### DISCUSSION TITLE
VCStool error and pip python discountinued

### DISCUSSION THREAD

**Discussion by KCHJ1999** [prl-mushr/mushr#69] [Software]
<img width="552" alt="vcstool" src="https://user-images.githubusercontent.com/96906767/147813016-88f11db3-95c1-442c-af22-10979e16489c.png">

Hi, I am a student doing my final year project this semester and my project topic is to replicate a MuSHR car but I came across some errors. my objective is to run rviz on my machine but when I was following the quickstart tutorial step, i came across an issue regarding the vcstool export repository.  

pip python 2.7 was also discontinued in January 2021 and no longer supports Python 2.7 so I am currently stuck at the repository side. is there any other method I could use so I could import the repository? Could someone tell me what does the error means?Thank You.

P.s I am not so knowledgeable regarding Linux and ROS, still learning. just trying to get things to work. Much help would be appreciated.

<img width="815" alt="cmd" src="https://user-images.githubusercontent.com/96906767/147813188-05b6fb3d-3489-499f-b51c-0ca294c38bd7.png">

<img width="626" alt="pip" src="https://user-images.githubusercontent.com/96906767/147813557-1968895d-f821-4855-8154-ce429b04cf13.png">

**Comment by palanc** [comment]
Hi @KCHJ1999 - Did you install vcstool using pip? If so, could you uninstall it using pip, install it through apt, and then try the vcs command again? I believe these commands will do it:

```sudo pip uninstall vcstool```
```sudo apt install python3-vcstool```
```cd ~/catkin_ws/src```
```vcs import < repos.yaml```

**Reply by KCHJ1999** [reply]
Hello, thanks for replying. Am i able to use python3 even though i am using a python 2.7 vcstool? will it have any compatibility issues?

**Reply by palanc** [reply]
If you follow the commands I listed I think that it will work. Have you given it a try?

**Reply by KCHJ1999** [reply]
I did another method but thank you for helping. :)

**Reply by KCHJ1999** [reply]
Erm can i ask what is the min voltage and current is
 needed to run the jetson and the peripherals(Realsense,YDLIDAR,VESC) because once i teleop my car, the jetson prompt me a low input voltage. Thank you.

**Reply by palanc** [reply]
Glad you were able to get it to work! For the low voltage issue, please refer to this post: https://github.com/prl-mushr/mushr/discussions/51

Let us know if that doesn't help.

**Reply by KCHJ1999** [reply]
Thank you. I was thinking of trying a 6000Mah Battery and 7.4V to power up the jetson and the peripherals. My jetson works fine but the only issue I have is the peripherals, it is consuming too much power. I will let u know if it works. I will try the method you sent me too. Thank you.

**Reply by KCHJ1999** [reply]
Hi, i increased the buck converter to 5.2V and now the jetson isnt prompting me low input voltage when i run the car! Thank you.

**Reply by palanc** [reply]
Awesome, glad that fixed it!
